packages/form-validations/src/FormValidations.ts

```diff
@@ -0,0 +1,496 @@
+const currentLocale = 'en';
+const validators: Record<string, Validator> = {};
+
+type Validator = {
+  fn: (value: string) => boolean;
+  name?: string;
+  priority: number;
+  halt: boolean;
+  msg?: string;
+};
+
+type Message = {
+  [locale: string]: {
+    required: string;
+    default: string;
+  };
+};
+
+type Field = {
+  errorElements?: HTMLElement[];
+  errors?: string[];
+  input: HTMLInputElement;
+  messages: Message[];
+  params: any;
+  validators: Validator[];
+};
+
+interface FormValidationsHTMLElement extends HTMLInputElement {
+  formValidations: Field;
+}
+
+type HTMLAttribute = {
+  name: string;
+  value: string;
+};
+
+const validate = (name: string, validator: Validator) => {
+  validator.name = name;
+
+  if (validator.priority === undefined) {
+    validator.priority = 1;
+  }
+
+  validators[name] = validator;
+};
+
+// validate('text', { fn: (val) => true, priority: 0 });
+validate('required', {
+  fn(value) {
+    return this.type === 'radio' || this.type === 'checkbox'
+      ? groupedElemCount(this)
+      : value !== undefined && value.trim() !== '';
+  },
+  priority: 99,
+  halt: true,
+});
+// validate('email', { fn: (val) => !val || EMAIL_REGEX.test(val)});
+// validate('number', { fn: (val) => !val || !isNaN(parseFloat(val)), priority: 2 });
+// validate('integer', { fn: (val) => !val || /^\d+$/.test(val) });
+// validate('minlength', { fn: (val, length) => !val || val.length >= parseInt(length) });
+// validate('maxlength', { fn: (val, length) => !val || val.length <= parseInt(length) });
+// validate('min', { fn: function(val, limit){ return !val || (this.type === 'checkbox' ? groupedElemCount(this) >= parseInt(limit) : parseFloat(val) >= parseFloat(limit)); } });
+// validate('max', { fn: function(val, limit){ return !val || (this.type === 'checkbox' ? groupedElemCount(this) <= parseInt(limit) : parseFloat(val) <= parseFloat(limit)); } });
+// validate('pattern', { fn: (val, pattern) => { let m = pattern.match(new RegExp('^/(.*?)/([gimy]*)$')); return !val || (new RegExp(m[1], m[2])).test(val);} });
+// validate('equals', { fn: (val, otherFieldSelector) => { let other = document.querySelector(otherFieldSelector); return (other) && ((!val && !other.value) || (other.value === val)); } });
+
+class FormValidations {
+  config: Config;
+  form: HTMLElement;
+  live: boolean;
+  fields: Field[];
+
+  constructor(form: HTMLElement, config: Config, live: boolean) {
+    form.setAttribute('novalidate', 'true');
+
+    this.form = form;
+    this.config = mergeConfig(config || {}, defaultConfig) as Config;
+    this.live = !(live === false);
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore Type 'Element' is missing the following properties from type 'HTMLInputElement': accept, align, alt, autocomplete, and 173 more.
+    this.fields = Array.from(form.querySelectorAll(SELECTOR)).map((input) => {
```

Comment:
Wouldn't it be better to split this function somehow? Alternatively, define a function outside the constructor and just call it in the constructor? By dividing it, you could better define the type that the method returns and then you could get rid of the ugly eslint comment

Reply:
Just moved to a separate method. I think I should split the logic into smaller chunks anyway.